microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/jaxrs/java/JaxRsWorkspaceSymbolParticipant.java

```diff
@@ -0,0 +1,246 @@
+import org.eclipse.lsp4mp.jdt.core.jaxrs.JaxRsContext;
+import org.eclipse.lsp4mp.jdt.core.jaxrs.JaxRsUtils;
+import org.eclipse.lsp4mp.jdt.core.utils.AnnotationUtils;
+import org.eclipse.lsp4mp.jdt.core.utils.IJDTUtils;
+import org.eclipse.lsp4mp.jdt.internal.jaxrs.JaxRsConstants;
+import org.eclipse.lsp4mp.jdt.internal.restclient.MicroProfileRestClientConstants;
+
+/**
+ * Collects workspace symbols for JAX-RS REST endpoints.
+ */
+public class JaxRsWorkspaceSymbolParticipant implements IJavaWorkspaceSymbolsParticipant {
+
+	private static final Logger LOGGER = Logger.getLogger(JaxRsWorkspaceSymbolParticipant.class.getName());
+
+	private static final SearchPattern SEARCH_PATTERN;
+	static {
+		SearchPattern leftPattern = null;
+		for (String annotation : JaxRsConstants.HTTP_METHOD_ANNOTATIONS) {
+			if (leftPattern == null) {
+				leftPattern = annotationSearchPattern(annotation);
+			} else {
+				leftPattern = SearchPattern.createOrPattern(leftPattern, annotationSearchPattern(annotation));
+			}
+		}
+		SEARCH_PATTERN = leftPattern;
+	}
+
+	@Override
+	public void collectSymbols(IJavaProject project, IJDTUtils utils, List<SymbolInformation> symbols, IProgressMonitor monitor) {
+		String applicationPrefix = getJaxApplicationPath(project);
+		SearchEngine engine = new SearchEngine();
+		IJavaSearchScope scope = BasicSearchEngine.createJavaSearchScope(true, new IJavaElement[] { project },
+				IJavaSearchScope.SOURCES);
+		Set<IAnnotatable> annotatables = new HashSet<>();
+		try {
+			engine.search(SEARCH_PATTERN, new SearchParticipant[] { SearchEngine.getDefaultSearchParticipant() }, scope,
+					new SearchRequestor() {
+
+						@Override
+						public void acceptSearchMatch(SearchMatch match) throws CoreException {
+							if (match.isInsideDocComment()) {
+								return;
+							}
+
+							if (match.getElement() instanceof IAnnotation) {
+								annotatables.add((IAnnotatable) ((IAnnotation) match.getElement()).getParent());
+							} else if (match.getElement() instanceof IAnnotatable) {
+								annotatables.add((IAnnotatable) match.getElement());
+							}
+						}
+
+					}, null);
+		} catch (CoreException | ClassCastException e) {
+			LOGGER.log(Level.SEVERE,
+					"While collecting symbols for project " + project.getResource().getLocationURI().toString(), e);
+		}
+		annotatables
+				.forEach(annotatable -> collectSymbolFromAnnotatable(symbols, annotatable, applicationPrefix, utils));
+	}
+
+	private void collectSymbolFromAnnotatable(List<SymbolInformation> symbols, IAnnotatable annotatable,
+			String applicationPrefix, IJDTUtils utils) {
+
+		if (!(annotatable instanceof IMethod)) {
+			return;
+		}
+
+		IMethod method = (IMethod) annotatable;
+		try {
+			if (!Flags.isPublic(method.getFlags())) {
+				return;
+			}
+		} catch (JavaModelException e) {
+			return;
+		}
+
+		String endpointUri = getEndpointUriFromAnnotatable(annotatable, applicationPrefix);
```

Comment:
I tried to play with your PR with https://github.com/FroMage/quarkus-renarde-todo/blob/main/src/main/java/rest/Todos.java and it doesn't work. 

It seems it is because methods doesn't use @Path although the method is annotated with @POST. If there are no @Path, we should use the method name if I remember.

Reply:
Ok let's see that in an another PR.

Reply:
I took a look at how Renard figures out which methods are REST endpoints, and it looks like any public method in a class that implements io.quarkiverse.renarde.Controller counts as a REST endpoint. Generally, the logic for calculating the method name looks different enough that I think it's worthwhile to provide a mechanism for quarkus-ls to disable the default WorkspaceSymbol support when it detects Renarde, and instead use the Renard support contributed through quarkus-ls.